### PR TITLE
[BEAM-8348] set job_name in portable_runner.py job request

### PR DESCRIPTION
**Pull request by ibzib**
Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/python/apache_beam/runners/portability/portable_runner.py [hunk 1]
```diff
@@ -296,7 +297,8 @@ def add_runner_options(parser):
 
     prepare_response = job_service.Prepare(
         beam_job_api_pb2.PrepareJobRequest(
-            job_name='job', pipeline=proto_pipeline,
+            job_name=options.view_as(GoogleCloudOptions).job_name or 'job',
```
Comment (tweise) [inline sdks/python/apache_beam/runners/portability/portable_runner.py]: GoogleCloudOptions doesn't seem appropriate for this. Should we relocate the option?

Comment (ibzib) [reply sdks/python/apache_beam/runners/portability/portable_runner.py]: That was initially my preferred approach, but I don't see a way to do it without breaking all the existing Dataflow pipelines that set a job name. See #9724

Comment (tweise) [reply sdks/python/apache_beam/runners/portability/portable_runner.py]: Would it be possible to replicate the option and deprecate it in `GoogleCloudOptions`?

FWIW this option also exists in the Flink runner and the job_name is already reflected in the pipeline (see https://github.com/apache/beam/pull/9752#issuecomment-541250003), so what is the effect of this change going to be?

Comment (ibzib) [reply sdks/python/apache_beam/runners/portability/portable_runner.py]: `PipelineOptions` does not allow the same option to belong to multiple subclasses.

This PR changes `job_name` in the job request, not the pipeline options. e.g. here https://github.com/apache/beam/blob/master/runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/InMemoryJobService.java#L124

Comment (tweise) [reply sdks/python/apache_beam/runners/portability/portable_runner.py]: I see, what is the user facing impact?

Regarding `PipelineOptions`, can it not be changed to allow for the co-existence of `job_name` in both classes for a transition period? @robertwb

Comment (robertwb) [reply sdks/python/apache_beam/runners/portability/portable_runner.py]: Yes, we could add an explicit `@property` in `GoogleCloudOptions` that delegates to the one in `PipelineOptions`.

Comment (ibzib) [reply sdks/python/apache_beam/runners/portability/portable_runner.py]: It's a good idea. Unfortunately, it looks like the override of `__settatr__` would take precedence over the property's setter. I'm not too familiar with how all these Python internals work, though, so I could be missing something obvious?

https://github.com/apache/beam/blob/16b4afa4469ee2d2fd5a5f4aafd77a156ee14e7b/sdks/python/apache_beam/options/pipeline_options.py#L336

Comment (robertwb) [reply sdks/python/apache_beam/runners/portability/portable_runner.py]: Huh. Perhaps override `_visible_option_list`?

**Comment by robertwb** [comment]
Any update on this?

**Comment by ibzib** [comment]
I haven't made any progress on this. It's not a terribly important feature.

**Pull request closed without merging**